credoai/reporting/model_fairness.py

```diff
@@ -412,4 +411,111 @@ def _plot_text(self, f, text_objects):
         ax.patch.set_alpha(0.01)
         for x, y, s, kwargs in text_objects:
             ax.text(x, y, s, transform = ax.transAxes, ha='center', **kwargs)
-        return ax
\ No newline at end of file
+        return ax
+
+
+class RegressionReporter(FairnessReporter):
+    def __init__(self, assessment, size=3):
+        super().__init__(assessment, size)
+    
+    def plot_results(self, filename=None):
+        """Creates a disaggregated performancereport for regression model
+
+        Parameters
+        ----------
+        filename : string, optional
+            If given, the location where the generated pdf report will be saved, by default None
+            
+        Returns
+        -------
+        array of figures
+        """        
+        df = self.module.get_df()
+        self.figs.append(self.plot_fairness())
+        self._plot_true_vs_pred_scatter(df, 'Disaggregated Performance')
+
+        plt.show()
+
+        if filename is not None:
+            self.export_report(filename)
+
+        return self.figs
+
+    def _create_report_cells(self):
+        # report cells
+        cells = [
+            self._write_fairness(),
+            ("reporter.plot_fairness();", 'code'),
+            self._write_true_vs_pred_scatter(),
+            ("""\
+            df = reporter.module.get_df()
+            reporter._plot_true_vs_pred_scatter(df, 'Disaggregated Performance')
+            """, 'code')
+        ]
+        return cells
+
+    def _plot_true_vs_pred_scatter(self, df, label):
+        """generates disaggregated scatter plot
+
+        Parameters
+        ----------
+        df : pandas.dataframe
+            dataframe of true and predicted values, including also the associated demographic groups
+        label : str
+            plot title
+
+        Returns
+        -------
+        matplotlib figure
+        """        
+        y_true, y_pred = df['true'], df['pred']
```

Comment:
If we have 10,000 datapoints, how does this plot fair? I'd probably place an upper limit on the number of data points by sampling without replacement.

Reply:
Good idea. This is now applied.